runtime/ceylon/language/Integer.java

```diff
@@ -420,7 +421,7 @@ public double getFloat() {
     public static double getFloat(long value) {
         if (value >= 9007199254740992L
                 || value <= -9007199254740992L) {
-            throw new OverflowException();
+            throw new OverflowException(value + " cannot be coerced into a 64 bit floating point value");
```

Comment:
I added here the exception message, like for Float.getInteger.

Actually, I'm not sure that this overflow check here is sensible. The class documentation says:

> An integer is considered equal to its `float` representation. That is, for every integer `int`, the expression `int.float==int` evaluates to `true`.

The presence of this exception here means that there are integers (and actually quite the majority, at least in the JVM) where this expression throws an OverflowException instead of evaluating to true (even including some which actually are representable exactly as Float).

Is this a good idea? Or should we instead compare `(long)(double)value` with `value` to see if there are problems? Or we could simply ignore any lost precision here. This would have the effect that several different `Integer` values can map to the same `Float` value, and for those cases `==` actually becomes non-transitive:

    assert(922337203685477580 == 922337203685477580.0);
    assert(922337203685477581 == 922337203685477580.0);
    assert(922337203685477581 != 922337203685477580);